hypershift-operator/controllers/nodepool/capi.go

```diff
@@ -428,6 +428,13 @@ func (c *CAPI) reconcileMachineDeployment(ctx context.Context, log logr.Logger,
 		},
 	}
 
+	// The CAPI provider for OpenStack uses the FailureDomain field to set the availability zone.
```

Comment:
@mdbooth when time permits, I'll like you eyes on that.

Some platforms have FailureDomain e.g. AzureMachine.Spec.FailureDomain, we don't have that in OpenStackMachine. We rely on the CAPI Machine.Spec.FailureDomain.
That's why I did it this way.